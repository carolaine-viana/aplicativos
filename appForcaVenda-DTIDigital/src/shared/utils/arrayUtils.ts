/**
 * Use this to deal/manipulate arrays
 */
export function isArrayEmpty<T = null>(arr: Array<T> | null | undefined) {
  return arr && arr?.length === 0;
}

export function doesArrayHaveContent<T = null>(arr: Array<T> | null | undefined) {
  const isEmpty = isArrayEmpty(arr);
  if (isEmpty) {
    return false;
  }
  return arr?.some((i) => !!i);
}

export function filterArraObjectAnyValue<T>(
  arr: Array<T>,
  valueFilter: string,
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  keysFilter?: Array<keyof T>
) {
  return arr.filter((item: T) => {
    const values = Object.values(item).map((value: string) => {
      if (typeof value !== 'object') {
        return String(value).toUpperCase();
      }
      return null;
    });
    return values.join(' ').includes(valueFilter.toUpperCase());
  });
}

export const addNoRepeatedString = (list: string[], newValue: string) => {
  const newList = list.filter((str) => str !== newValue || str.length === 0);
  newList.push(newValue);
  return newList;
};

export const mergArrayIntoStrings = (list: string[], separator = ';') => {
  if (!list || isArrayEmpty(list)) {
    return '';
  }

  const noEmptyStrings = list.filter((str) => str.length > 0);
  const mergedString =
    noEmptyStrings.length === 1 ? noEmptyStrings[0] : noEmptyStrings.join(separator);

  return mergedString;
};
