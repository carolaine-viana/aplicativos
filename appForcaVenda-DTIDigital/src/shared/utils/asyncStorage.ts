import AsyncStorage from '@react-native-community/async-storage';
import { Profiles, TRole } from '@shared/types';
import { addNoRepeatedString, getISOStringDate, mergArrayIntoStrings } from '.';

export const FV_TOKEN = '@fv/token';
export const FV_TOKEN_EXPIRATION = '@fv/tokenExpiration';
export const FV_USER_LOGIN = '@fv/rcaLogin';
export const FV_USER_ROLE = '@fv/userRole';
export const FV_FIRST_TUTO_MAP = '@fv/firstTutoMap';
export const FV_FEATURE_FLAGS = '@fv/featureFlags';
export const FV_COMPLETED_TUTORIALS = '@fv/completedTutorials';
export const FV_LAST_TIME_USER_SAW_TUTORIAL = '@fv/lastTimeUserSawTutorial';
export const FV_DATE_ACCEPTED_TERM_USE = '@fv/dateAcceptedTermUse';
export const FV_DATE_UPDATED_TERM_USE = '@fv/dateUpdatedTermUse';

export const clear = () => AsyncStorage.clear();

// TOKEN INFO
export const setToken = async (token: string) => {
  await AsyncStorage.setItem(FV_TOKEN, token);
};

export const getToken = async () => {
  const token = await AsyncStorage.getItem(FV_TOKEN);
  return token;
};

export const setTokenExpiration = async (expirationDate: Date) => {
  await AsyncStorage.setItem(FV_TOKEN_EXPIRATION, expirationDate.toString());
};

export const getTokenExpiration = async (): Promise<Date | number> => {
  const expirationTime = await AsyncStorage.getItem(FV_TOKEN_EXPIRATION);
  return expirationTime ? new Date(expirationTime) : 0;
};

// USER INFO
export const getUserLogin = async () => {
  const userLogin = await AsyncStorage.getItem(FV_USER_LOGIN);
  return userLogin;
};

export const setUserLogin = async (login: string) => {
  await AsyncStorage.setItem(FV_USER_LOGIN, login);
};

export const getUserRole = async (): Promise<TRole | null> => {
  const userRole = await AsyncStorage.getItem(FV_USER_ROLE);
  return userRole as TRole;
};

export const setUserRole = async (role: TRole) => {
  await AsyncStorage.setItem(FV_USER_ROLE, role);
};

export const isUserGV = async (): Promise<boolean> => {
  const userRole = await getUserRole();
  return userRole === Profiles.GV;
};

// FEATURE FLAG
export const setFeatureFlags = async (featureFlags: string) => {
  await AsyncStorage.setItem(FV_FEATURE_FLAGS, featureFlags);
};

export const getFeatureFlags = async () => {
  const flags = await AsyncStorage.getItem(FV_FEATURE_FLAGS);
  return flags || '';
};

// TUTORIALS
export const getCompletedTutorials = async () => {
  const completedTutorials = await AsyncStorage.getItem(FV_COMPLETED_TUTORIALS);
  return completedTutorials ?? '';
};

export const setCompletedTutorials = async (tutorial: string) => {
  const completedTutorials = await getCompletedTutorials();
  if (completedTutorials !== ';') {
    const newListCompletedTutorials = completedTutorials.split(';');
    const uniqueStrings = addNoRepeatedString(newListCompletedTutorials, tutorial);
    await AsyncStorage.setItem(FV_COMPLETED_TUTORIALS, mergArrayIntoStrings(uniqueStrings));
  }
};

export const getLastTimeUserSawTutorial = () =>
  AsyncStorage.getItem(FV_LAST_TIME_USER_SAW_TUTORIAL);

export const setLastTimeUserSawTutorial = () => {
  const isoStringDate = getISOStringDate(new Date());
  AsyncStorage.setItem(FV_LAST_TIME_USER_SAW_TUTORIAL, isoStringDate);
};

export const getFirstTutoMap = async (visualized?: string) => {
  if (visualized === 'visualized') {
    await AsyncStorage.setItem(FV_FIRST_TUTO_MAP, 'visualized');
  }
  const firstMap = await AsyncStorage.getItem(FV_FIRST_TUTO_MAP);
  return firstMap;
};

// PRIVACY POLICY
export const getAcceptedTermUseDate = () => AsyncStorage.getItem(FV_DATE_ACCEPTED_TERM_USE);

export const setDateAcceptedTermUse = (lastDate: string | undefined) => {
  if (lastDate) {
    return AsyncStorage.setItem(FV_DATE_ACCEPTED_TERM_USE, lastDate);
  }
};

export const getUpdatedTermUseDate = () => AsyncStorage.getItem(FV_DATE_UPDATED_TERM_USE);
export const setDateUpdatedTermUse = (updatedDate: string | undefined) => {
  if (updatedDate) {
    return AsyncStorage.setItem(FV_DATE_UPDATED_TERM_USE, updatedDate);
  }
};
