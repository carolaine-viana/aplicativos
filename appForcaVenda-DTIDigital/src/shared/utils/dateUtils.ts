/**
 * Use this to deal/manipulate dates
 */
import * as dateFNS from 'date-fns';
import pt from 'date-fns/locale/pt-BR';
import { zonedTimeToUtc } from 'date-fns-tz';

interface IDateUtils {
  getTimeZonedDate: (date: Date) => Date;
  formatDatePtBr: (date: Date, format?: string) => string;
  formatDateOnlyPtBr: (date: Date, format?: string) => string;
  formatHour: (date: Date, format?: string) => string;
}

const getTimeZonedDate: IDateUtils['getTimeZonedDate'] = (date: Date) =>
  zonedTimeToUtc(date, 'ptBR');

const formatDateOnlyPtBr: IDateUtils['formatDateOnlyPtBr'] = (
  date: Date,
  format = " dd 'de' MMMM "
) =>
  dateFNS.format(date, format, {
    locale: pt,
  });

const formatHour: IDateUtils['formatHour'] = (date: Date, format = "HH:mm'h'") =>
  dateFNS.format(date, format);

const formatDate = (date: Date, format = 'dd-MM-yyyy') => dateFNS.format(date, format);

const getUnixTime = (date: Date) => dateFNS.getUnixTime(date);
const getDateFromUnix = (dateNumber: number) => dateFNS.fromUnixTime(dateNumber);

const getDateWithTime = (date: Date, h: string | number, m: string | number) => {
  const dateWithTime = dateFNS.setHours(date, Number(h));
  dateWithTime.setMinutes(Number(m || 0));
  return dateWithTime;
};

const getNumberDayOfWeek = (date: Date) => dateFNS.getDay(date);

const getLastMonday = (fromDate: Date) => {
  let lastMondayDate = fromDate;
  while (!dateFNS.isMonday(lastMondayDate)) {
    lastMondayDate = dateFNS.subDays(lastMondayDate, 1);
  }
  return lastMondayDate;
};

const addDays = (fromDate = new Date(), amount = 0) => dateFNS.addDays(fromDate, amount);

const addMinutes = (fromDate: Date | number, amount: number) =>
  dateFNS.addMinutes(fromDate, amount);

const getStartOfMonth = (date: Date) => getTimeZonedDate(dateFNS.startOfMonth(date));

const getEndOfMonth = (date: Date) => getTimeZonedDate(dateFNS.endOfMonth(date));

const mountTokenExpirationDate = (tokenExpirationSeconds: number) => {
  const offSetMinutes = 5;
  const now = new Date();
  const time = addMinutes(now, tokenExpirationSeconds / 60 - offSetMinutes);
  return time;
};

const getYearMonth = (
  date = new Date()
): { monthYearNumber: string; monthYearDescription: string } => ({
  monthYearNumber: dateFNS.format(date, 'yyyyMM'),
  monthYearDescription: dateFNS.format(date, "MMMM' 'yyyy", {
    locale: pt,
  }),
});

const getMonthDescription = (month: number = 0, format = 'MMM') => {
  const year = new Date().getFullYear();
  return dateFNS.format(new Date(Number(year), Number(month)), format, {
    locale: pt,
  });
};

const subtractMonths = (date = new Date(), amountMonths: number) =>
  dateFNS.subMonths(date, amountMonths);

const getISOStringDate = (date: Date) => date.toISOString();

const getDiffHoursFromDate = (fromDate: Date, toDate: Date) => {
  const subtractHours = dateFNS.differenceInHours(fromDate, toDate);
  return subtractHours;
};

const getLastMonthsDate = (
  amountMonths: number,
  includeCurrentMonth: boolean = false,
  formatDateString: string = 'MMMM yyyy'
) => {
  let currentDate = new Date();
  const lastMonthsDates: { stringDate: string; pureDate: Date }[] = [];

  for (let index = amountMonths; index > 0; index--) {
    const lastMonth = subtractMonths(currentDate, 1);
    const formatedDate = dateFNS.format(lastMonth, formatDateString, {
      locale: pt,
    });
    lastMonthsDates.push({ stringDate: formatedDate, pureDate: lastMonth });
    currentDate = lastMonth;
  }

  if (includeCurrentMonth) {
    const today = new Date();
    lastMonthsDates.unshift({
      stringDate: dateFNS.format(today, formatDateString, {
        locale: pt,
      }),
      pureDate: today,
    });
  }

  return lastMonthsDates;
};

export {
  formatDate,
  getUnixTime,
  getDateFromUnix,
  getDateWithTime,
  formatHour,
  formatDateOnlyPtBr,
  getNumberDayOfWeek,
  getLastMonday,
  addDays,
  addMinutes,
  getStartOfMonth,
  getEndOfMonth,
  getTimeZonedDate,
  mountTokenExpirationDate,
  getYearMonth,
  getMonthDescription,
  subtractMonths,
  getDiffHoursFromDate,
  getISOStringDate,
  getLastMonthsDate,
};
