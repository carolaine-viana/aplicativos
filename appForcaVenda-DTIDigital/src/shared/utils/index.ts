export * from './generalUtils';
export * from './dateUtils';
export * from './arrayUtils';
export * from './mappers/mapperCustomerVisit';
