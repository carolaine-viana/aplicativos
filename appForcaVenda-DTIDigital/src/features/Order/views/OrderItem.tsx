import React from 'react';

import { DefaultIcon, Typography } from '@shared/components';
import { getCurrency } from '@shared/utils';
import { Container, Content, Info, InfoDate } from './OrderItemSC';

interface ItemProps {
  date: string;
  orderNumber: string;
  value: number;
}

interface OrderItemProps {
  item: ItemProps;
  onPress: () => void;
}
const OrderItem: React.FC<OrderItemProps> = ({ item, onPress }) => (
  <Container onPress={onPress}>
    <Content>
      <InfoDate>
        <Typography variant="overline" color="gray300" size={11}>
          {item.date}
        </Typography>
      </InfoDate>

      <Info>
        <Typography variant="leadMedium" size={16}>{`Pedido: ${item.orderNumber}`}</Typography>
      </Info>
      <Info>
        <Typography variant="bodyMedium" size={14} color="blue600">
          {getCurrency(item.value)}
        </Typography>
      </Info>
    </Content>
    <DefaultIcon name="chevron-right" size={24} color="primary" />
  </Container>
);

export default OrderItem;
