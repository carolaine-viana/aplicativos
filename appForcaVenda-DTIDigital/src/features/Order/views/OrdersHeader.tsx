import React from 'react';

import { formatCNPJ } from '@shared/utils';
import { ClientCNPJ, ClientName } from '../../cardClient/views/CardClientSC';
import { Container } from './OrdersHeaderSC';

interface Props {
  id: string;
  cnpj: string;
  tradeName: string;
  clientName: string;
}

interface HeaderProps {
  customer: Props;
}
const OrdersHeader: React.FC<HeaderProps> = ({ customer }) => (
  <Container>
    <ClientCNPJ variant="bodyMedium" color="gray800" bold>
      {`${formatCNPJ(customer.cnpj)} • ${customer.id}`}
    </ClientCNPJ>
    <ClientName variant="overline" color={customer.tradeName ? 'default' : 'gray500'}>
      {customer.tradeName || 'NOME FANTASIA NÃO ENCONTRADO'}
    </ClientName>
    <ClientName variant="overline">{customer.clientName}</ClientName>
  </Container>
);

export default OrdersHeader;
