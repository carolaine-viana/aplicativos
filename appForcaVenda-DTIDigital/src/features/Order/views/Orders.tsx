import React, { useState } from 'react';
import { useTheme } from 'styled-components';
import { CustomStatusBar, Divider, SearchHeader } from '@shared/components';
import { useNavigation } from '@react-navigation/native';
import { Container } from './OrdersSC';
import OrderItem from './OrderItem';
import OrdersHeader from './OrdersHeader';

const item = {
  date: '17/01/2022',
  orderNumber: '0987654',
  value: 200,
};

const Orders: React.FC = ({ route }) => {
  const theme = useTheme();
  const navigation = useNavigation();

  const [search, setSearch] = useState('');

  const handleCloseSearch = () => {
    setTimeout(() => {
      setSearch('');
    }, 200);
  };

  return (
    <>
      <CustomStatusBar backgroundColor={theme.palette.blue50} barStyle="dark-content" />
      <SearchHeader
        headerProps={{ title: 'Selecione o pedido' }}
        onClickArrow={() => navigation.goBack()}
        searchProps={{
          onChangeText: (text: string) => {
            setSearch(text);
          },
          onClose: handleCloseSearch,
          value: search,
          searchPlaceholder: 'Buscar pedido',
        }}
      />
      <Container>
        <OrdersHeader customer={route?.params} />
        <Divider color="blue100" noMarge />
        <OrderItem item={item} onPress={() => {}} />
      </Container>
    </>
  );
};

export default Orders;
