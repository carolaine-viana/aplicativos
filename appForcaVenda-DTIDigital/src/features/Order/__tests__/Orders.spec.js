import React from 'react';
import expect from 'expect';
import { render} from '@shared/utils/test-utils';
import Orders from '../views/Orders'
import OrderItem from '../views/OrderItem'
import OrdersHeader from '../views/OrdersHeader'
import jestExpect from 'expect';
import * as navigationHooks from '@react-navigation/native';

jest.mock('@react-navigation/native', () => ({
  useNavigation: jest.fn(),
  useRoute: jest.fn(),
}));

jest.mock('styled-components', () => {
  return {
    useTheme: jest.fn().mockReturnValue({
      palette: {
        white: '#ffffff',
      },
    }),
  };
});


it('should be render OrderItem correctly', () => {
  const screen = render(<OrderItem item={{date: '', orderNumber: '', value: ''}}/>).toJSON();
  jestExpect(screen).toMatchSnapshot();
})

it('should be render OrdersHeader correctly', () => {
  const screen = render(<OrdersHeader customer={{id: '', cnpj: '', tradeName: '', clientName: ''}} />).toJSON();
  jestExpect(screen).toMatchSnapshot();
})


it.skip('should be render Orders correctly', () => {
  navigationHooks.useRoute.mockReturnValueOnce({
    id: '00',
    cnpj: '0000',
    tradeName: '',
    clientName: '',
  });
  const screen = render(<Orders />).toJSON();
  jestExpect(screen).toMatchSnapshot();
})



