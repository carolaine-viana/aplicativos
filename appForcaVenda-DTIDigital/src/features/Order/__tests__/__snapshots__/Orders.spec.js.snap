// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be render OrderItem correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginTop": 24,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "column",
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "marginBottom": 8,
          },
        ]
      }
    >
      <Text
        color="gray300"
        fallbackText=""
        size={11}
        style={
          Array [
            Object {
              "color": "#999DA9",
              "fontFamily": "Rubik-Regular",
              "fontSize": 11,
              "lineHeight": 13,
              "textTransform": "none",
            },
          ]
        }
        variant="overline"
      >
        
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "marginBottom": 4,
          },
        ]
      }
    >
      <Text
        color="default"
        fallbackText=""
        size={16}
        style={
          Array [
            Object {
              "color": "#10151D",
              "fontFamily": "Rubik-Medium",
              "fontSize": 16,
              "lineHeight": 19,
              "textTransform": "none",
            },
          ]
        }
        variant="leadMedium"
      >
        Pedido: 
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "marginBottom": 4,
          },
        ]
      }
    >
      <Text
        color="blue600"
        fallbackText=""
        size={14}
        style={
          Array [
            Object {
              "color": "#072AD8",
              "fontFamily": "Rubik-Medium",
              "fontSize": 14,
              "lineHeight": 17,
              "textTransform": "none",
            },
          ]
        }
        variant="bodyMedium"
      >
        R$ 0,00
      </Text>
    </View>
  </View>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#072AD8",
          "fontSize": 24,
        },
        undefined,
        Object {
          "fontFamily": "Feather",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`should be render Orders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginTop": 24,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "column",
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "marginBottom": 8,
          },
        ]
      }
    >
      <Text
        color="gray300"
        fallbackText=""
        size={11}
        style={
          Array [
            Object {
              "color": "#999DA9",
              "fontFamily": "Rubik-Regular",
              "fontSize": 11,
              "lineHeight": 13,
              "textTransform": "none",
            },
          ]
        }
        variant="overline"
      >
        
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "marginBottom": 4,
          },
        ]
      }
    >
      <Text
        color="default"
        fallbackText=""
        size={16}
        style={
          Array [
            Object {
              "color": "#10151D",
              "fontFamily": "Rubik-Medium",
              "fontSize": 16,
              "lineHeight": 19,
              "textTransform": "none",
            },
          ]
        }
        variant="leadMedium"
      >
        Pedido: 
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "marginBottom": 4,
          },
        ]
      }
    >
      <Text
        color="blue600"
        fallbackText=""
        size={14}
        style={
          Array [
            Object {
              "color": "#072AD8",
              "fontFamily": "Rubik-Medium",
              "fontSize": 14,
              "lineHeight": 17,
              "textTransform": "none",
            },
          ]
        }
        variant="bodyMedium"
      >
        R$ 0,00
      </Text>
    </View>
  </View>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#072AD8",
          "fontSize": 24,
        },
        undefined,
        Object {
          "fontFamily": "Feather",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`should be render Orders correctly 2`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginTop": 24,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "column",
        },
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "marginBottom": 8,
          },
        ]
      }
    >
      <Text
        color="gray300"
        fallbackText=""
        size={11}
        style={
          Array [
            Object {
              "color": "#999DA9",
              "fontFamily": "Rubik-Regular",
              "fontSize": 11,
              "lineHeight": 13,
              "textTransform": "none",
            },
          ]
        }
        variant="overline"
      >
        
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "marginBottom": 4,
          },
        ]
      }
    >
      <Text
        color="default"
        fallbackText=""
        size={16}
        style={
          Array [
            Object {
              "color": "#10151D",
              "fontFamily": "Rubik-Medium",
              "fontSize": 16,
              "lineHeight": 19,
              "textTransform": "none",
            },
          ]
        }
        variant="leadMedium"
      >
        Pedido: 
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "marginBottom": 4,
          },
        ]
      }
    >
      <Text
        color="blue600"
        fallbackText=""
        size={14}
        style={
          Array [
            Object {
              "color": "#072AD8",
              "fontFamily": "Rubik-Medium",
              "fontSize": 14,
              "lineHeight": 17,
              "textTransform": "none",
            },
          ]
        }
        variant="bodyMedium"
      >
        R$ 0,00
      </Text>
    </View>
  </View>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#072AD8",
          "fontSize": 24,
        },
        undefined,
        Object {
          "fontFamily": "Feather",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`should be render OrdersHeader correctly 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 8,
      },
    ]
  }
>
  <Text
    bold={true}
    color="gray800"
    fallbackText=""
    style={
      Array [
        Object {
          "color": "#19202D",
          "fontFamily": "Rubik-Bold",
          "fontSize": 14,
          "lineHeight": 17,
          "textTransform": "none",
        },
        Object {
          "marginBottom": 12,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 12,
        },
      ]
    }
    variant="bodyMedium"
  >
     • 
  </Text>
  <Text
    color="gray500"
    fallbackText=""
    style={
      Array [
        Object {
          "color": "#6D7384",
          "fontFamily": "Rubik-Regular",
          "fontSize": 11,
          "lineHeight": 13,
          "textTransform": "none",
        },
        Object {
          "fontSize": 12,
          "marginBottom": 12,
        },
      ]
    }
    variant="overline"
  >
    NOME FANTASIA NÃO ENCONTRADO
  </Text>
  <Text
    color="default"
    fallbackText=""
    style={
      Array [
        Object {
          "color": "#10151D",
          "fontFamily": "Rubik-Regular",
          "fontSize": 11,
          "lineHeight": 13,
          "textTransform": "none",
        },
        Object {
          "fontSize": 12,
          "marginBottom": 12,
        },
      ]
    }
    variant="overline"
  >
    
  </Text>
</View>
`;
